aten/src/ATen/core/ivalue_inl.h

```diff
@@ -355,6 +355,19 @@ struct C10_EXPORT ivalue::Future : c10::intrusive_ptr_target {
    * Explicitly mark the future as completed with the output value.
    */
   void markCompleted(IValue value) {
+    markCompletedWithDataPtrs(std::move(value));
+  }
+
+  /**
+   * Explicitly mark the future as comleted with the output value and DataPtrs.
+   * The data_ptrs contains storage pointers for all tensors in IValue, which
+   * will be passed to postMarkCompletedHook. Some subclass, like CUDAFuture,
+   * uses these DataPtrs to synchronize CUDA streams.
```

Comment:
nit: Maybe we should mention in the docs that we should use this API only if we can't extract tensors from the IValue itself.